.github/workflows/ci-cd.yml

```diff
@@ -71,18 +71,19 @@ jobs:
         run: pytest -v --markdown-docs -m markdown-docs modal
 
   container-dependencies:
-    name: Check minimal container dependencies for ${{ matrix.python-version }}
+    name: Check minimal container dependencies for ${{ matrix.python-version }} / ${{ matrix.image-builder-version }}
     runs-on: ubuntu-20.04
     timeout-minutes: 4
     strategy:
       matrix:
-        python-version: ["3.8", "3.9", "3.10", "3.11", "3.12"]
+        image-builder-version: ["2024.04", "2024.10"]
+        python-version: ["3.8", "3.12"]
```

Comment:
Yes — was worried about combinatorial explosion of this check. It will make debugging failures a little annoying, but I think the chances that a mandatory dependency gets added after 3.8 (soon to be 3.9) and then removed before 3.12 (soon to be 3.13) are pretty low, so risk of false negative should be rare?